Blaze: manage campaigns: use `can_blaze` from me sites endpoint to check for blaze eligibility

AjeshRPai opened the pull request:
part of #18704

## Description 
This PR updates the eligibility check of a site to enable the blaze feature. 

## Explanation of changes 
- THis PR removes the blaze status API dependancy for the blaze eligibility check for a site - which was added with the fluxc PR  https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2657/commits
- this pr implements the blaze feature to be enabled by checking the `can_blaze` property in site model 

## To test:
### test - blaze is not available

1. go to the **jetpack** app
2. 🚪 login with a wp account without administrator privileges or use a private site 
3. ⛳ go to **app settings** → **debug settings →** enable feature flag → `blaze`
4. 👉🏻go to dashboard
5. ✅ verify that the **🔥 promote with blaze** is not shown 

### test - blaze is available

1. go to the **jetpack** app
2. 🚪 login with a wp account with administrator privileges 
3. ⛳ go to **app settings** → **debug settings →** enable feature flag → `blaze`
4. 👉🏻go to dashboard
5. ✅ verify that the **🔥 promote with blaze** is shown

### Test - Blaze is not available

1. Go to the **Jetpack** app
2. 🚪 Login with a wp account without Administrator privileges or Use a private site 
3. 👉🏻Go to Pages → Published Pages 
4. 👆🏻Click on the more icon ⁞ → 
5. ✅ Verify that the **🔥 Promote with Blaze** is not shown 

### Test - Blaze is available

1. Go to the **Jetpack** app
2. 🚪 Login with a wp account with Administrator privileges 
3. 👉🏻Go to Pages → Published Pages 
4. 👆🏻Click on the more icon ⁞ → 
5. ✅ Verify that the **🔥 Promote with Blaze** is shown 

### Test - Blaze is not available
1. Go to the **Jetpack** app
2. 🚪 Login with a wp account without Administrator privileges or Use a private site 
3. ⛳ Go to **app settings** → **Debug settings →** enable feature flag → `blaze`
4. 👉🏻Go to Posts → Published Posts 
5. 👆🏻Click on the more icon ⁞ → 
6. ✅ Verify that the **🔥 Promote with Blaze** is not shown 

### Test - Blaze is available
1. Go to the **Jetpack** app
2. 🚪 Login with a wp account with Administrator privileges 
3. ⛳ Go to **app settings** → **Debug settings →** enable feature flag → `blaze`
4. 👉🏻Go to Posts → Published Posts 
5. 👆🏻Click on the more icon ⁞ → 
6. ✅ Verify that the **🔥 Promote with Blaze** is shown

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)


## Merge instructions 
- Make sure that https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2762 is merged 
- update the fluxC version 
- remote donot merge label 
- merge as normal

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

AjeshRPai committed:
+ adds: function to check blaze eligibility from site model attribute

AjeshRPai committed:
* updates: the function to check blaze eligibility of post

Updates the blaze eligibility check to use site model attribute

AjeshRPai committed:
- removes: redundant blaze eligibility api call in PostListMainViewModel

As the blaze eligibility is checked from site model, there is no need
to call blaze status api

AjeshRPai committed:
- removes: redundant blaze eligibility api call in PagesViewModel

As the blaze eligibility is checked from site model, there is no need
to call blaze status api

AjeshRPai committed:
- removes: redundant blaze eligibility api call in WpMainActivityVM

As the blaze eligibility is checked from site model, there is no need
to call blaze status api

AjeshRPai committed:
- removes: redundant blaze eligibility check

- removes: redundant blaze eligibility check
CreatePageListItemActionsUseCase as this check is already done when
passing the parameter `isPageEligibleForBlaze`

AjeshRPai committed:
* Updates: the logic in PromoteWithBlazeCardSource

+ Adds: the logic of checking whether the current site is eligible for
blaze from can blaze property in BlazeFeatureUtils
* Updates: the logic in PromoteWithBlazeCardSource to use the helper
 function in BlazeFeatureUtils
* Updates: the logic in PromoteWithBlazeCardSource to check for the
blaze manage campaigns feature
* Updates: the data class for passing the card update to the viewmodel
to remove site id and add campaigns to the class
* Updates: the logic in MySiteViewModel to use the updated data class

AjeshRPai committed:
- removes: redundant blaze dependencies and imports

- removes: the redundant imports
- removes: the redundant dependencies for checking blaze eligibility

AjeshRPai committed:
- removes: redundant imports

AjeshRPai committed:
* updates: the helper function for readability

+ adds: missing can blaze check for site
* updates: function to be private
* updates: the if else check for readablility and optimizes the check

AjeshRPai committed:
+ adds: blaze eligibility check for page and updates the usage

AjeshRPai committed:
* fixes: incorrect test logic

AjeshRPai committed:
* updates: logic of blaze with null check for can_blaze

AjeshRPai committed:
* updates: the fluxC version

zwarm reviewed (changes requested):
@AjeshRPai - The blaze tests all pass as written 👍 ; however I am seeing issues with the refreshing indicator 😿 

Pixel 5, Android 13
Steps to recreate
- Fresh login
- Select site
- Navigate to dashboard
- The refresh indicator is never-ending
- Tap from Home > Menu > Home, the refresh indicator hides
- The PTR function no longer works
- Change sites and the PTR still is not functioning

AjeshRPai committed:
* fixes: the PTR not updating

AjeshRPai commented:
Hey @zwarm 👋🏼 

Thanks for the review. The PTR issues has been fixed with the commit [acb4057](https://github.com/wordpress-mobile/WordPress-Android/pull/18718/commits/acb4057c44f133a611941dc3f5475ad715b448c0)

THe issue was that I was directly updating the mediator livedata without using the `postState` extension function which updates the `refresh` mediator live data value. 🤦🏼 . So have fixed the 
1. PTR refresh indicator never ending 
2. PTR not functioning

AjeshRPai committed:
* fixes: the refresh data being called twice

AjeshRPai committed:
* updates: the data fetch logic to use coroutine scope

AjeshRPai committed:
* fixes: the test statement logic

AjeshRPai committed:
* renames: PromoteWithBlazeCardSource and usages to BlazeCardSource

* renames PromoteWithBlazeCardSource.kt and
PromoteWithBlazeCardSourceTest.kt and usages to BlazeCardSource as
there will be more than one type of blaze card after introducing
campaigns

AjeshRPai commented:
Hey @zwarm 

On Investigating the unit tests failure I found that 
1. the refresh in BlazeCardSource.kt is called twice so have done some changes with the commit [c245fda](https://github.com/wordpress-mobile/WordPress-Android/pull/18718/commits/c245fda5a484bd485253118c22f141070af43d41)
2. fixed the unit test according to the above logic 
3. get data logic is not inside coroutine, so changed that with [16c0fe2](https://github.com/wordpress-mobile/WordPress-Android/pull/18718/commits/16c0fe21d4f67e3d12e0a3b2f40f6715a53125e8)

I renamed the `promotewithBlazecardsource` to `blazecardsource`, on re-visiting this PR i felt that this renaming could have been shifted to another PR. I am sorry about that.  😞

zwarm committed:
Merge branch 'trunk' into issue/18704-use-can_blaze-from-me-sites-endpoint-to-check-for-blaze-eligibility

zwarm commented on WordPress/src/main/java/org/wordpress/android/ui/blaze/BlazeFeatureUtils.kt:
```diff
@@ -46,7 +46,9 @@ class BlazeFeatureUtils @Inject constructor(
-        return siteModel.canBlaze && siteModel.isAdmin &&
+        return siteModel.canBlaze != null &&
+                siteModel.canBlaze &&
+                siteModel.isAdmin &&
```
@AjeshRPai - ❓ I thought we didn't need the isAdmin check, but it's quite likely that I am mistaken.

AjeshRPai replied:
Hey @zwarm 

I think We need the Admin check, checked IOS PR - [they have same logic](https://github.com/wordpress-mobile/WordPress-iOS/pull/20916/files). I think can_blaze returns true for contributor role as well, don't remember but I think I checked that as well. Feel free to test it again

zwarm reviewed (approved):
@AjeshRPai - Thanks for tackling this. All the tests pass as expected. I left one question, but it is not a blocker. 🙇 

Have to wait on the FluxC PR hash before merging.

zwarm committed:
Update FluxC hash
